src/wazuh_modules/syscollector/syscollector_windows.c

```diff
@@ -1007,42 +1118,50 @@ void sys_hw_windows(const char* LOCATION){
     cJSON_AddStringToObject(object, "timestamp", timestamp);
     cJSON_AddItemToObject(object, "inventory", hw_inventory);
 
-    /* Get Serial number */
+    /* Base Board Serial Number variables */
     char *serial = NULL;
-    memset(read_buff, 0, SERIAL_LENGTH);
-    command = "wmic baseboard get SerialNumber";
-    output = popen(command, "r");
-    if (!output){
-        mtwarn(WM_SYS_LOGTAG, "Unable to execute command '%s'.", command);
-    }else{
-        if (fgets(read_buff, SERIAL_LENGTH, output)) {
-            if (strncmp(read_buff ,"SerialNumber", 12) == 0) {
-                if (!fgets(read_buff, SERIAL_LENGTH, output)){
-                    mtwarn(WM_SYS_LOGTAG, "Unable to get Motherboard Serial Number.");
-                    serial = strdup("unknown");
-                }
-                else if (end = strpbrk(read_buff,"\r\n"), end) {
-                    *end = '\0';
-                    int i = strlen(read_buff) - 1;
-                    while(read_buff[i] == 32){
-                        read_buff[i] = '\0';
-                        i--;
-                    }
-                    serial = strdup(read_buff);
-                }else
-                    serial = strdup("unknown");
-            }
-        } else {
-            mtdebug1(WM_SYS_LOGTAG, "Unable to get Motherboard Serial Number (bad header).");
-            serial = strdup("unknown");
-        }
-
-        if (status = pclose(output), status) {
-            mtwarn(WM_SYS_LOGTAG, "Command 'wmic' returned %d getting board serial.", status);
-        }
-    }
-
-    cJSON_AddStringToObject(hw_inventory, "board_serial", serial);
+	DWORD smbios_size = 0;
+	PRawSMBIOSData smbios = NULL;
+	
+	DWORD Signature = 0;
+	const BYTE byteSignature[] = { 'B', 'M', 'S', 'R' }; // "RSMB" (little endian)
+	memcpy(&Signature, byteSignature, 4);
+	
+	/* Get raw SMBIOS firmware table size */
+	/* Reference: https://docs.microsoft.com/en-us/windows/desktop/api/sysinfoapi/nf-sysinfoapi-getsystemfirmwaretable */
+	smbios_size = GetSystemFirmwareTable(Signature, 0, NULL, 0);
```

Comment:
This function is not compatible with versions older than Windows Vista. This makes impossible to install the agent on Windows XP/Server 2003 which is needed. I suggest to include the call to this function to `syscollector_win_ext.dll` as other functions with the same issue like `inet_ntop`.

Reply:
Fixed in 886edafb9a0212b7a6e4720ccafdb85b09f8e9d2.